src/bootstrap/compile.rs

```diff
@@ -1444,12 +1447,13 @@ pub fn run_cargo(
         };
         for filename in filenames {
             // Skip files like executables
-            if !(filename.ends_with(".rlib")
+            if !((!rlib_only_metadata && filename.ends_with(".rlib"))
```

Comment:
This is very hard for me to understand. Could you do something like `(rlib_only_metadata && filename.ends_with(".rlib")) || !( // ...` instead?

Reply:
I split it into multiple if's and added comments. Is this less confusing?

Reply:
Looks a lot better, thank you!